src/env0-environments-provider.ts

```diff
@@ -42,6 +42,7 @@ export class Env0EnvironmentsProvider
   implements vscode.TreeDataProvider<Environment>
 {
   private environments: EnvironmentModel[] = [];
+  private isRefreshing = false;
```

Comment:
I wonder if in the context of VSCode extension using private var as a locking/syncing mechanism is the right choice, especially in case of [Activation Events](https://code.visualstudio.com/api/references/activation-events).
Did you consider other alternative? Maybe using a [data storage](https://code.visualstudio.com/api/extension-capabilities/common-capabilities#data-storage)?

Reply:
I'm sorry I'm not sure I understand how it's related to activation events and data storage? 
I use isRefreshing to ensure that when a refresh is triggered—either by clicking or through `setInterval` the refresh actions don't occur concurrently

Reply:
`isRefreshing` is used as a lock to the section between lines 102 and 114.
Using a private var as lock assumes that all threads of the extensions will have access to the same value.
My question is whether this assumption is correct in case the extension (or part of it) was executed using Activation Events.
Data storage is an alternative way to store the state of the lock (`isRefreshing`).

Reply:
The `isRefreshing` doesn't have to be thread safe as there is no scenarios where multiple instances of the extension running, therefore using a local var as a lock works.